stageleft/src/runtime_support.rs

```diff
@@ -75,11 +75,29 @@ impl ParseFromLiteral for bool {
 impl_parse_from_literal_numeric!(i8, i16, i32, i64, i128, isize);
 impl_parse_from_literal_numeric!(u8, u16, u32, u64, u128, usize);
 
-pub trait FreeVariable<O> {
-    fn to_tokens(self) -> (Option<TokenStream>, Option<TokenStream>)
+pub trait FreeVariableWithContext<Ctx> {
```

Comment:
(Maybe default `Ctx=()`?)

Reply:
We want free variables to be as generic wrt contexts as possible, so this forces developers to do that.